node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -30,8 +30,10 @@ object Main {
 
     val exec: Task[Unit] =
       for {
-        conf     <- Configuration(args)
-        poolSize = conf.server.threadPoolSize
+        conf <- Configuration(args)
+        poolSize <- conf.server.threadPoolSize.fold(Task.delay {
+                     Runtime.getRuntime().availableProcessors() * 2
+                   })(_.pure[Task])
         //TODO create separate scheduler for casper
```

Comment:
@lukasz-golebiewski mentioned that availableProcessors * 2 is not good heuristic, comments welcomed

Reply:
This is a good heuristic AFAIK. *2 if the task is I/O bound but we should test of course.

Reply:
It's better than 4000 threads IMO.

Reply:
Is this thread pool used for CPU intensive work or IO? 
https://www.ibm.com/developerworks/library/j-jtp0730/index.html